diagnostics.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Contains all configured adapters for the given environment.
+ *
+ * @type {Array}
+ * @public
+ */
+var adapters = [];
```

Comment:
why the `var` over `let`/`const`. What browser support are you aiming for here?

Reply:
As this is the "base" file that every environment depends upon I wanted to ensure that its at least ES5 compatible to support a wide varity of browser. If I where to use `let` or `const` that would mean that I would have to add `babel` compilation for the browser version.

Reply:
for es5 yes, all evergreen browsers support let/const though. Personally I'm slowly moving towards not giving a fuck about non evergreen browsers.